src/action/finder/request.rs

```diff
@@ -0,0 +1,1355 @@
+) -> Result<Vec<Condition>, ParseError> {
+    let mut conditions: Vec<Condition> = vec![];
+    if let Some(options) = proto {
+        if let Some(micros) = options.min_last_access_time {
+            let time = time_from_micros(micros)?;
+            conditions.push(Condition::MinAccessTime(time));
+        }
+        if let Some(micros) = options.max_last_access_time {
+            let time = time_from_micros(micros)?;
+            conditions.push(Condition::MaxAccessTime(time));
+        }
+    }
+    Ok(conditions)
+}
+
+fn get_inode_change_time_conditions(
+    proto: Option<FileFinderInodeChangeTimeCondition>,
+) -> Result<Vec<Condition>, ParseError> {
+    let mut conditions: Vec<Condition> = vec![];
+    if let Some(options) = proto {
+        if let Some(micros) = options.min_last_inode_change_time {
+            let time = time_from_micros(micros)?;
+            conditions.push(Condition::MinInodeChangeTime(time));
+        }
+        if let Some(micros) = options.max_last_inode_change_time {
+            let time = time_from_micros(micros)?;
+            conditions.push(Condition::MaxInodeChangeTime(time));
+        }
+    }
+    Ok(conditions)
+}
+
+fn get_size_conditions(
+    proto: Option<FileFinderSizeCondition>,
+) -> Vec<Condition> {
+    let mut conditions: Vec<Condition> = vec![];
+    if let Some(options) = proto {
+        if let Some(size) = options.min_file_size {
+            conditions.push(Condition::MinSize(size));
+        }
+        if options.max_file_size() < u64::MAX {
+            conditions.push(Condition::MaxSize(options.max_file_size()));
+        }
+    }
+    conditions
+}
+
+fn get_ext_flags_condition(
+    proto: Option<FileFinderExtFlagsCondition>,
+) -> Vec<Condition> {
+    let mut conditions: Vec<Condition> = vec![];
+    if let Some(options) = proto {
+        if let Some(bits) = options.linux_bits_set {
+            conditions.push(Condition::ExtFlagsLinuxBitsSet(bits));
+        }
+        if let Some(bits) = options.linux_bits_unset {
+            conditions.push(Condition::ExtFlagsLinuxBitsUnset(bits));
+        }
+        if let Some(bits) = options.osx_bits_set {
+            conditions.push(Condition::ExtFlagsOsxBitsSet(bits));
+        }
+        if let Some(bits) = options.osx_bits_unset {
+            conditions.push(Condition::ExtFlagsOsxBitsUnset(bits));
+        }
+    }
+    conditions
+}
+
+fn parse_regex(bytes: Vec<u8>) -> Result<Regex, ParseError> {
+    let str = match std::str::from_utf8(bytes.as_slice()) {
+        Ok(v) => Ok(v),
+        Err(e) => Err(ParseError::Malformed(Box::new(e))),
+    }?;
+
+    match Regex::new(str) {
+        Ok(v) => Ok(v),
+        Err(e) => Err(RegexParseError {
+            raw_data: bytes,
+            error: e,
+        }
+        .into()),
+    }
+}
+
+fn get_contents_regex_match_condition(
+    proto: Option<FileFinderContentsRegexMatchCondition>,
+) -> Result<Vec<Condition>, ParseError> {
+    if let Some(options) = proto {
```

Comment:
You can reduce nesting by inverting the condition:

```rust
let options = match proto {
    Some(options) => options,
    None => return Ok(vec!())`,
}
```

Reply:
Done!